### PR TITLE
[BUILD] Adding multi-architecture images for arm32 v7 and amd64/x64

### PR DESCRIPTION
**Pull request by AVSurfer123**
Converted the Dockerfiles so that they use the multi-architecture image for the `latest` tag. This way, you can run the same tag on either `arm32` or `amd64`. I had to update travis.yml with the new commands to create these multi-architecture images. I also
moved the repository from `avsurfer123/c-runtime` to `pierobotics/c-runtime` to be more official.

Sidenotes regarding Travis CI: I moved/changed the repository environment variables in the online travis-ci.org settings https://travis-ci.org/github/pioneers/c-runtime/settings. These include

- DOCKER_USERNAME: the username to login with, currently its mine (avsurfer123) but we might want to change it
- DOCKER_ACCESS_TOKEN: the access token used to login in place of my password
- DOCKER_REPO: The repo to push the images to, currently `pierobotics/c-runtime`
- TAG: the tag to use for the multi-architecture images of `master`, currently `latest`

### CODE REVIEW

File: docker/README.md [hunk 1]
```diff
@@ -10,47 +10,46 @@ A Docker **image** is a compressed object holding all the dependencies that you
...
 A **Dockerfile** is how you specify the dependencies in your image. It uses a simplified command structure (see https://docs.docker.com/engine/reference/builder/) which allows you to run any bash command. The nice thing about a Docker image is that it caches the image between every `RUN`, `ADD` or `COPY` command. This allows you to build images quickly after you've built it the first time. 
 
-Our latest Docker images will be stored on Docker Hub here https://hub.docker.com/repository/docker/avsurfer123/c-runtime.
+Our latest Docker images will be stored on Docker Hub here https://hub.docker.com/repository/docker/pierobotics/c-runtime.
 
 
 ## Installation
 
-To install Docker, go here https://docs.docker.com/get-docker/.
+To install Docker, go here https://docs.docker.com/get-docker/. If you're on Windows and don't have Windows 10 Pro, it is highly recommended to install WSL 2 here https://docs.microsoft.com/en-us/windows/wsl/install-win10.
```
Comment (levincent06) [inline docker/README.md:18]: What does Windows 10 Pro offer that doesn't really need WSL? What does this have to do with Docker? Does this have to do with architecture?

Comment (AVSurfer123) [reply docker/README.md:18]: Docker Desktop can't be normally installed if you have Windows 10 Home, I think due to HyperV issues. If you have Home, you either need to install Docker Toolbox which has Oracle VirtualBox, which is not recommended, or install WSL 2 to have a Linux VM.

File: docker/README.md [hunk 2]
```diff
@@ -10,47 +10,46 @@ A Docker **image** is a compressed object holding all the dependencies that you
...
 
 ## Pulling
 
-To get the image to emulate a Raspberry Pi, do `docker pull avsurfer123/c-runtime:latest`. This will have the code from the `master` branch.
+To get the image to emulate a Raspberry Pi, do `docker pull pierobotics/c-runtime:latest`. This will have the code from the `master` branch.
 
 ## Building
```
[the remark this answers is not in the document]
Comment (AVSurfer123) [reply docker/README.md:24]: yea. i can add a section for that

File: docker/README.md [hunk 3]
```diff
@@ -10,47 +10,46 @@ A Docker **image** is a compressed object holding all the dependencies that you
...
+If you haven't run this already on your machine and so don't have local caches, it will take several hours. Also, this can only be run on x64 machines. If you are using Linux, you will also need to install QEMU before building by doing `docker run --rm --privileged multiarch/qemu-user-static --reset -p yes`.
 
 ## Running
 
-To run the latest Runtime inside a Docker container, do `docker run -it --rm avsurfer123/c-runtime:latest`. Then you can test by in another terminal doing `docker exec -it $(docker ps -q) bash` and running whatever you want inside the container. The `c-runtime` repo will be located at `/root/runtime`.
+To run the latest Runtime inside a Docker container, do `docker run -it --rm pierobotics/c-runtime:latest`. Then you can test by in another terminal doing `docker exec -it $(docker ps -q) bash` and running whatever you want inside the container. The `c-runtime` repo will be located at `/root/runtime`.
 
 ### Devices
 
 If you want to have the container access any Arduino devices, add the flag `--device /dev/ttyACM0` or what ever devices you want to `docker run`.
```
[the remark this answers is not in the document]
Comment (AVSurfer123) [reply docker/README.md:50]: I mean no one should be using Windows without WSL lol. On WSL its a different port name but same idea

File: docker/README.md [hunk 4]
```diff
@@ -10,47 +10,46 @@ A Docker **image** is a compressed object holding all the dependencies that you
...
+    DOCKER_CLI_EXPERIMENTAL=enabled docker buildx build --platform linux/amd64,linux/arm/v7 --build-arg BUILDKIT_INLINE_CACHE=1 --cache-from pierobotics/c-runtime:base -t pierobotics/c-runtime:base --push .
+
+If you haven't run this already on your machine and so don't have local caches, it will take several hours. Also, this can only be run on x64 machines. If you are using Linux, you will also need to install QEMU before building by doing `docker run --rm --privileged multiarch/qemu-user-static --reset -p yes`.
 
 ## Running
 
-To run the latest Runtime inside a Docker container, do `docker run -it --rm avsurfer123/c-runtime:latest`. Then you can test by in another terminal doing `docker exec -it $(docker ps -q) bash` and running whatever you want inside the container. The `c-runtime` repo will be located at `/root/runtime`.
+To run the latest Runtime inside a Docker container, do `docker run -it --rm pierobotics/c-runtime:latest`. Then you can test by in another terminal doing `docker exec -it $(docker ps -q) bash` and running whatever you want inside the container. The `c-runtime` repo will be located at `/root/runtime`.
```
[the remark this answers is not in the document]
Comment (AVSurfer123) [reply docker/README.md:46]: Yea I can do that. Is the Windows command on WSL? You said you don't use WSL right?

File: docker/README.md [hunk 5]
```diff
@@ -10,47 +10,46 @@ A Docker **image** is a compressed object holding all the dependencies that you
...
 
 ## Building
 
 To build your own image instead of using the one on Docker Hub, so that it uses your up to date code instead of code on `master`, do
     
-    DOCKER_BUILDKIT=1 docker build --build-arg BUILDKIT_INLINE_CACHE=1 -t avsurfer123/c-runtime:latest -f docker/Dockerfile .
+    docker build -t pierobotics/c-runtime:latest -f docker/Dockerfile .
 
-### Base Image
+If you're on a x64 machine and have the ability to emulate ARM32 either by using Docker Desktop on Windows/Mac or have installed QEMU on Linux (with `docker run --rm --privileged multiarch/qemu-user-static --reset -p yes`), you can instead build an ARM32 image with
```
[the remark this answers is not in the document]
Comment (AVSurfer123) [reply docker/README.md:30]: Linux doesn't have Docker Desktop. Only Mac/Windows do. Docker Desktop provides QEMU built-in, so on Linux you need to install it yourself. I can make this more clear.

File: .travis.yml [hunk 6]
```diff
@@ -1,36 +1,50 @@
...
+        - curl -fsSL https://download.docker.com/linux/ubuntu/gpg | sudo apt-key add -
+        - sudo add-apt-repository "deb [arch=amd64] https://download.docker.com/linux/ubuntu $(lsb_release -cs) edge"
+        - sudo apt-get update
+        - sudo apt-get -y -o Dpkg::Options::="--force-confnew" install docker-ce
+      script:
+        # DOCKER_USERNAME and DOCKER_ACCESS_TOKEN are defined as a repository secret environment variable on travis-ci.org
+        - echo "$DOCKER_ACCESS_TOKEN" | docker login -u "$DOCKER_USERNAME" --password-stdin
+        - docker run --rm --privileged multiarch/qemu-user-static --reset -p yes  # Need to install QEMU to emulate ARM architectures
+        - docker buildx create --use
+        - docker buildx build --platform linux/amd64,linux/arm/v7 -t $DOCKER_REPO:$TAG --build-arg BUILDKIT_INLINE_CACHE=1 --push -f docker/Dockerfile .
```
[the remark this answers is not in the document]
Comment (AVSurfer123) [reply .travis.yml:50]: The `push` jobs is the thing that builds the multi-architecture images. Basically the only reason I kept the `docker push` in the `test` jobs is for the rare case where you're on `x64` but you want to pull the `arm32` image. Currently that's not possible using Docker's multi-arch images and you would have to build the `arm32` image yourself on your `x64` machine. So this skips the 3 min it takes to build the `arm32` image on `x64`. Do you think I should keep this, or remove this tag?

File: docker/README.md (hunk 4 above)
[the remark this answers is not in the document]
Comment (levincent06) [reply docker/README.md:46]: I don't use WSL, and that Windows command works fine for me

File: docker/README.md (hunk 3 above)
[the remark this answers is not in the document]
Comment (levincent06) [reply docker/README.md:50]: I'm using Windows without WSL and I've been doing fine :  ) I have it installed but I've been using a VM instead, then switched over to Docker and still haven't had a need to use it

What happens if you plug in an Arduino to Windows? Because dev handler searches specifically for ports with files matching the pattern `/dev/ttyACM`

File: .travis.yml (hunk 6 above)
[the remark this answers is not in the document]
Comment (AVSurfer123) [reply .travis.yml:44]: I ended up removing it because this is only for advanced users, who should know how to rebuild the image anyways.

File: docker/README.md (hunk 3 above)
[the remark this answers is not in the document]
Comment (AVSurfer123) [reply docker/README.md]: On WSL, the devices appear at `/dev/ttyS*`
